src/views/SingleMovie.vue

```diff
@@ -0,0 +1,148 @@
+<script setup lang="ts">
+import { computed } from "vue";
+import BreadCrumb from "@/components/BreadCrumb.vue";
+import { useMovieStore } from "@/stores/movies";
+import { useRoute } from "vue-router";
+import lengthToTime from "@/helpers/lengthToTime";
+import type { Movie } from "@/types/movie"
+const movieStore = useMovieStore();
+const route = useRoute();
+
+const selectedId = computed((): string => route.params.id.toString());
+const selectedMovie = computed(
+  (): Movie =>
+    movieStore.movieList.find(
+      (movie) => movie.id.toString() === selectedId.value
+    ) || []
+);
```

Comment:
If you want to provide a ype for your computed you shuld do this like so:
```js
const selectedMovie = computed<Movie>(() => {
  // ...
});
```

Also, you either return a movie or... an  empty array 🤔 This will lead to errors if we can't find the movie in the store 😄

Reply:
So i take it is better practice to let it rerun a`Movie | undefined' and deal with 'undefined' at component/ui level?

Reply:
Yup! Or we might do a `find(...) || null` if we do not like undefined but both are fine 👍